### PR TITLE
Collect information to help track down a crash.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Useful for figuring out what's causing: https://github.com/dotnet/roslyn/issues/63732

### CODE REVIEW

File: src/Features/Core/Portable/Wrapping/Edit.cs [hunk 1]
```diff
@@ -63,9 +65,27 @@ public static Edit DeleteBetween(SyntaxNodeOrToken left, SyntaxNodeOrToken right
...
+        {
+            // Used for analyzing dumps
+#pragma warning disable IDE0052 // Remove unread private members
+            private readonly SyntaxTree? _tree;
+            private readonly SyntaxToken _left;
+            private readonly SyntaxToken _right;
+#pragma warning restore IDE0052 // Remove unread private members
+
+            public InvalidEditException(SyntaxToken left, SyntaxToken right)
+                : base($"Left token: '{left}' had an end past the start of right token: '{right}'")
```
Comment (ryzngard) [inline src/Features/Core/Portable/Wrapping/Edit.cs:83]: if we can't keep trees, can we actually keep tokens in the message?

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/Wrapping/Edit.cs:83]: @jasonmalinowski this should pii information right?

Comment (CyrusNajmabadi) [inline src/Features/Core/Portable/Wrapping/Edit.cs:83]:
```suggestion
                : base($"Left token had an end '{left.Span.End}' past the start of right token '{right.Span.Start}'")
```

**Comment by CyrusNajmabadi** [comment]
@ryzngard Switched to just passing ints.  Hopefully that's ok.

**Review by ryzngard — APPROVED** [review]

**Pull request merged**